autopush/main.py

```diff
@@ -467,7 +467,8 @@ def endpoint_main(sysargs=None, use_files=True):
 
     # Endpoint HTTP router
     site = cyclone.web.Application([
-        (r"/push/([^\/]+)", EndpointHandler, dict(ap_settings=settings)),
+        (r"/push/(v\d+)?\/?([^\/]+)", EndpointHandler,
```

Comment:
The / should be part of the optional bit. The re module has a non-capturing () which should do just fine rather than the extra ? on the trailing /, ie:
```
r"/push/(?:(v\d)/)?([^/]+)"
```